src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs

```diff
@@ -290,7 +290,8 @@ private static string GetHome()
             }
             else
             {
-                return GetValueOrThrowMissingEnvVarsDotnet(() => GetDotNetHome() ?? Environment.GetEnvironmentVariable(Home), Home, DotNetHome);
+                return GetValueOrThrowMissingEnvVarsDotnet(() =>
+                    GetDotNetHome() ?? Environment.GetEnvironmentVariable(Home) ?? Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), Home, DotNetHome);
```

Comment:
`SpecialFolder` didn't exist in .NET Core 1.0, which is why I guess NuGet's custom home finding method was added.  But since the API exists now, I think we should delete it and only use the .NET API.

This might be a breaking change, which is why our PR template asks for a linked issue in NuGet/Home. Those issues get linked in our release notes, and raise awareness of such changes.

Frankly I'm astonished that containers exist where the HOME environment variable isn't set.  Surely NuGet isn't the only app that doesn't expect this.  Why doesn't Docker make it a pre-defined variable, rather than trying to get all other tools to change?

Reply:
> sorry, but i really don't understand how is it a breaking change?

Are you 100% certain that, for the cases that an exception is not thrown, that `Environment.GetFolderPath(SpecialFolder.UserProfile)` will always return the same value that NuGet's current code will return?

If there is any scenario at all where the old code might return a different value than the new code, then to somebody, somewhere, it's probably a breaking change.

Even if there's no breaking change, it's still something good to call out in our release notes. I wish our release notes were based on pull requests, rather than issues, but the engineering effort to change our process and tools isn't worth the benefit, so it's just easier and quicker to create a quick issue in Home and then link it in pull reuqest, just like our template asks you to twice:

https://github.com/NuGet/NuGet.Client/blob/0cccbcb539f692c3f1fdf1eff217f22993d2e0f2/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L3-L8

https://github.com/NuGet/NuGet.Client/blob/0cccbcb539f692c3f1fdf1eff217f22993d2e0f2/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L14